misc/helm-charts/operator/README.md

```diff
@@ -346,6 +347,122 @@ kubectl logs -l app.kubernetes.io/name=materialize-operator -n materialize
 
 For more detailed information on using and troubleshooting the Materialize operator, refer to the [Materialize documentation](https://materialize.com/docs).
 
+## Upgrading Materialize
+
+Once you have the Materialize operator installed and managing your Materialize instances, you can upgrade both components. While the operator and instances can be upgraded independently, you should ensure version compatibility between them. The operator can typically manage instances within a certain version range - upgrading the operator too far ahead of your instances may cause compatibility issues.
+
+We recommend:
+
+- Upgrade the operator first
+- Always upgrade your Materialize instances after upgrading the operator to ensure compatibility
+
+### Upgrading the Helm Chart
+
+To upgrade the Materialize operator to a new version:
+
+```shell
+helm upgrade my-materialize-operator materialize/misc/helm-charts/operator
+```
+
+If you have custom values, make sure to include your values file:
+
+```shell
+helm upgrade my-materialize-operator materialize/misc/helm-charts/operator -f my-values.yaml
+```
+
+### Upgrading Materialize Instances
+
+To upgrade your Materialize instances, you'll need to update the Materialize custom resource and trigger a rollout.
+
+By default, the operator performs rolling upgrades (`inPlaceRollout: false`) which minimize downtime but require additional cluster resources during the transition. For environments where scheduled downtime is acceptable, you can opt for in-place upgrades (`inPlaceRollout: true`).
```

Comment:
> For environments where scheduled downtime is acceptable

nit: I'd scratch `scheduled`, since we're not doing any scheduling here / users might not need it to be on a schedule.

I think it's also worth mentioning that rolling upgrades take longer